src/RpcServer/RpcServer.SmartContract.cs

```diff
@@ -144,6 +144,65 @@ protected virtual JObject InvokeScript(JArray _params)
             return GetInvokeResult(script, signers);
         }
 
+        [RpcMethod]
+        protected virtual JObject InvokeIterator(JArray _params)
+        {
+            UInt160 script_hash = UInt160.Parse(_params[0].AsString());
+            string operation = _params[1].AsString();
+            ContractParameter[] args = _params.Count >= 3 ? ((JArray)_params[2]).Select(p => ContractParameter.FromJson(p)).ToArray() : Array.Empty<ContractParameter>();
+            Signers signers = _params.Count >= 4 ? SignersFromJson((JArray)_params[3], system.Settings) : null;
+            uint? startIndex = _params.Count >= 5 ? Convert.ToUInt32(_params[4].AsNumber()) : null;
+            uint? endIndex = _params.Count >= 6 ? Convert.ToUInt32(_params[5].AsNumber()) : null;
+
+            byte[] script;
+            using ScriptBuilder sb = new();
+            script = sb.EmitDynamicCall(script_hash, operation, args).ToArray();
+
+            Transaction tx = signers == null ? null : new Transaction
+            {
+                Signers = signers.GetSigners(),
+                Attributes = Array.Empty<TransactionAttribute>(),
+                Witnesses = signers.Witnesses,
+            };
+            using ApplicationEngine engine = ApplicationEngine.Run(script, system.StoreView, container: tx, settings: system.Settings, gas: settings.MaxGasInvoke);
+            JObject json = new();
+            json["script"] = Convert.ToBase64String(script);
+            json["state"] = engine.State;
+            json["gasconsumed"] = engine.GasConsumed.ToString();
+            json["exception"] = GetExceptionMessage(engine.FaultException);
+            List<object> resultList = new();
+            try
+            {
+                var result = engine.ResultStack.FirstOrDefault() as VM.Types.InteropInterface;
+                if (result is null)
+                {
+                    json["result"] = null;
+                }
+                else
+                {
+                    var iterator = result.GetInterface<IIterator>();
+                    for (int i = 0; iterator.Next(); i++)
+                    {
+                        if (i >= startIndex && i < endIndex)
+                            resultList.Add(iterator.Value());
+                    }
+                    json["result"] = new JArray(resultList.Select(p => ((VM.Types.StackItem)p).ToJson()));
+                }
+                json["startindex"] = startIndex;
+                json["endindex"] = endIndex;
```

Comment:
It will be the same as the request, we can remove them

Reply:
> It will be the same as the request, we can remove them

It's in use for hyper result. (for prevent malicious attack)